### PR TITLE
Add configurations to "General" specialization

### PR DESCRIPTION
**Pull request by hansonchun-quartech**
### Description

This PR makes it so that a "General" specialization is now configurable. A "General" specialization is now just a regular specialization but without an enrichment index. The previous logic that returned a Default general specialization has been removed. Therefore, a general specialization will need to be created in each deployed environment.

When creating a specialization on the UI, the data source options will only be available when an enrichment index is selected.

### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

**Comment by mattthiessendev** [comment]
I have some concerns for this proposal. This is a lot of logic to add for something that is meant to be part of the system. We always expect the "General" specialization to be available to the user. Why not just have it be part of the system from the beginning? We control the rules for our data containers. Why not have the "General" specialization be populated in out database on initialization of the container itself? We can still add some logic in the backend to protect it and some frontend logic to display that specialization differently to the user.

**Comment by wdavidcamden** [comment]
> I have some concerns for this proposal. This is a lot of logic to add for something that is meant to be part of the system. We always expect the "General" specialization to be available to the user. Why not just have it be part of the system from the beginning? We control the rules for our data containers. Why not have the "General" specialization be populated in out database on initialization of the container itself? We can still add some logic in the backend to protect it and some frontend logic to display that specialization differently to the user.

we actually might need to eventually move to a scenario where someone doesn't have access to General ....

File: webapi/Services/QSpecializationService.cs [hunk 1]
```diff
@@ -105,11 +112,14 @@ public async Task<Specialization> SaveSpecialization(QSpecializationMutate qSpec
     }
 
     /// <summary>
-    /// Updates the specialization.
+    /// Updates an existing specialization.
     /// </summary>
-    /// <param name="specializationId">Unique identifier of the specialization</param>
-    /// <param name="qSpecializationMutate">Specialization mutate payload</param>
-    /// <returns>The task result contains the specialization source</returns>
+    /// <param name="specializationId">Identifier of the specialization to update or create.</param>
```
Comment (mattthiessendev) [inline webapi/Services/QSpecializationService.cs:117]: docs: this method will not create a new entity

File: webapp/src/components/admin/specialization/SpecializationManager.tsx [hunk 2]
```diff
@@ -231,11 +235,25 @@ export const SpecializationManager: React.FC = () => {
         }
     }, [editMode, selectedId, specializations]);
 
+    useEffect(() => {
+        const isValid =
+            !!label && !!name && !!roleInformation && !!description && !!initialChatMessage && membershipId.length > 0;
+        setIsValid(isValid);
+        return () => {};
```
Comment (mattthiessendev) [inline webapp/src/components/admin/specialization/SpecializationManager.tsx:242]: question: what is this doing?

**Review by mattthiessendev — APPROVED** [review]
todo: in a future PR we need to remove the `Choose Specialization` workflow for a new chat session if the user has no specializations to pick from

File: webapp/src/components/admin/specialization/SpecializationManager.tsx (hunk 2 above)
Comment (hansonchun-quartech) [reply webapp/src/components/admin/specialization/SpecializationManager.tsx:242]: This checks whether or not the form is valid. If those fields are not populated, the form is invalid and the Save button is disabled.

**Review by mattthiessendev — APPROVED** [review]

**Pull request merged**